packages/shared/components/StakingAssetTile.svelte

```diff
@@ -0,0 +1,246 @@
+        getUnstakedFunds,
+        hasAccountReachedMinimumAirdrop,
+        isAccountStaked,
+        isStakingPossible,
+    } from 'shared/lib/participation/staking'
+    import {
+        assemblyStakingRemainingTime,
+        partiallyStakedAccounts,
+        participationOverview,
+        shimmerStakingRemainingTime,
+        stakedAccounts,
+        stakingEventState,
+    } from 'shared/lib/participation/stores'
+    import { ParticipationEventState, StakingAirdrop } from 'shared/lib/participation/types'
+    import { WalletAccount } from 'shared/lib/typings/wallet'
+    import { openPopup } from 'shared/lib/popup'
+    import { getBestTimeDuration } from 'shared/lib/time'
+    import { formatUnitBestMatch } from 'shared/lib/units'
+    import { capitalize } from 'shared/lib/utils'
+    import { selectedAccount } from 'shared/lib/wallet'
+
+    export let asset: Asset
+
+    type TooltipText = {
+        title: string
+        body: string[]
+    }
+
+    const airdrop = asset?.name === Token.Assembly ? StakingAirdrop.Assembly : StakingAirdrop.Shimmer
+
+    let isBelowMinimumRewards: boolean
+    let canShowTooltip = false
+    let tooltipAnchor: HTMLElement
+    let tooltipText: TooltipText
+    let remainingTime: number
+
+    $: assetIconColor = isBright(asset?.color) ? 'gray-800' : 'white'
+    $: isDarkModeEnabled = $appSettings.darkMode
+    $: isActivelyStaking = _getAccount($stakedAccounts) && isStakingPossible($stakingEventState)
+    $: isPartiallyStaked = _getAccount($partiallyStakedAccounts) && isStakingPossible($stakingEventState)
+    $: hasStakingEnded = $stakingEventState === ParticipationEventState.Ended
+    $: $participationOverview, (tooltipText = getLocalizedTooltipText())
+    $: remainingTime = asset?.name === Token.Assembly ? $assemblyStakingRemainingTime : $shimmerStakingRemainingTime
+    $: {
+        if (hasAccountReachedMinimumAirdrop($selectedAccount) && !isStakingPossible($stakingEventState)) {
+            isBelowMinimumRewards = false
+        } else {
+            const accountOverview = $participationOverview.find(
+                (accountParticipationOverview) => accountParticipationOverview.accountIndex === $selectedAccount?.index
+            )
+            isBelowMinimumRewards = accountOverview?.[`${asset?.name}RewardsBelowMinimum`] > 0
+        }
+    }
+    $: canShowWarningState =
+        isPartiallyStaked ||
+        (isBelowMinimumRewards && !_getAccount($stakedAccounts) && isStakingPossible($stakingEventState)) ||
+        (isBelowMinimumRewards && hasStakingEnded)
+
+    function toggleTooltip(): void {
+        canShowTooltip = !canShowTooltip
+    }
+    function handleTileClick(): void {
+        openPopup({ type: 'airdropNetworkInfo', props: { airdrop } })
+    }
+    function _getAccount(accounts: WalletAccount[]): WalletAccount {
+        return accounts.find((account) => account.alias === $selectedAccount.alias)
+    }
+    function getLocalizedTooltipText(): TooltipText {
```

Comment:
This function needs refactoring. It's very difficult to follow. Try writing it with only one layer of if's and without elses or closures. I think they add unnecessary complexity.

Reply:
If you want we can have a look at it together

Reply:
I've copied [this code](https://github.com/iotaledger/firefly/blob/develop/packages/shared/components/AccountTile.svelte#L110-L222) and done some refactoring, but yeah I will improve it more

Reply:
Will refactor it later, we need this PR merged ASAP